libraries/type-length-value/src/state.rs

```diff
@@ -408,6 +408,54 @@ impl<'a> TlvState for TlvStateMut<'a> {
     }
 }
 
+/// Packs a borsh-serializable value into an existing TLV space, reallocating
+/// the account and TLV as needed to accommodate for any change in space
+#[cfg(feature = "borsh")]
+pub fn realloc_and_borsh_serialize<V: TlvDiscriminator + borsh::BorshSerialize>(
+    account_info: &solana_program::account_info::AccountInfo,
```

Comment:
Should this not just operate over a slice of bytes, rather than an `AccountInfo`? 
All of the other methods in the lib don't use any structs from `solana-program`, which allows this library to be used generically over any buffers

I see you're offering some ease of use for realloc, but that may not belong in this library

Reply:
I see this as a common pattern for any program that wants to keep its accounts at the minimum required size. Most of this library assumes fixed-size types defined through `Pod`, so this is definitely a different situation. You're correct -- even the borsh examples just operate on slices of bytes.

I originally had this in the token-metadata program, but since the order of operations is tricky ie. do I realloc TLV first or the underlying account? And you don't want to get this wrong, or you can potentially clobber your TLV and render your account unusable.

Since this is generic for any borsh types, we can provide a working implementation for any program to use. Do you have any ideas of where it could go?

Reply:
I'll see if I can come up with a better home for borsh/`AccountInfo` stuff, but for now I think it's cool to merge. I can address later in another PR